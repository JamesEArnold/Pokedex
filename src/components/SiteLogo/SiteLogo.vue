<template>
  <div>
    <img
      :src="require(`../../assets/${logoUrl}`)"
      alt="Site Logo"
      class="logo--main"
    >
  </div>
</template>

<script lang="ts">
import { defineComponent } from 'vue';

export default defineComponent({
  name: 'SiteLogo',
  props: {
    logoUrl: {
      type: String,
      required: true,
    },
  },
});
</script>

<style lang="scss">
.logo {
  &--main {
    max-width: 30vw;
    min-width: 240px;
  }
}
</style>
