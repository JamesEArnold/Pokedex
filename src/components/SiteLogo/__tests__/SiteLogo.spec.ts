import { VueWrapper, mount } from '@vue/test-utils';
import { ComponentPublicInstance } from 'vue';
import SiteLogo from '../SiteLogo.vue';

describe('SiteLogo.vue', () => {
  let wrapper: VueWrapper<ComponentPublicInstance<typeof SiteLogo>>;
  const someLogo = 'logo.png';

  beforeEach(() => {
    expect.hasAssertions();
    wrapper = mount(SiteLogo, {
      props: {
        logoUrl: someLogo,
      },
    });
  });

  it('should render component with logourl prop', () => {
    expect(wrapper.vm.logoUrl).toEqual(someLogo);
  });

  it('has the expected html structure', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
