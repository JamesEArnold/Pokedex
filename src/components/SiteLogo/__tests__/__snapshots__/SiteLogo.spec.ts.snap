// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteLogo.vue has the expected html structure 1`] = `
<div>
  <img
    alt="Site Logo"
    class="logo--main"
    src=""
  />
</div>
`;

